### PR TITLE
Error page components

### LINKED ISSUE #47: Add error report page
As a data entry person, after using the Dropzone to upload a CSV, I want to be automatically directed to an error report for my data with the option to print or download, so that I can easily identify corrections that I need to make (either at my desk or tomorrow in the field) before re-uploading a revised file.

- [x] #39

### PR DESCRIPTION
**Pull request by gabriellaarcilla**
Added error report table. Read data from a .json file and displayed on table. Added error messages.
fixes https://github.com/ForestGeoHack/ForestGEO/issues/47

### CODE REVIEW

**Comment by OmarShehata** [comment]
@gabriellaarcilla this looks great! Quick reminder: when opening a PR please link to the original GitHub issue to add context & to make it easier to track our tasks.

To add an error message to specific rows, try this:

* Change `errorMessage` from a string to an object
* It should contain a map of `rowNumber -> error string`

So to insert an error for row 3 it would look like this:

```
const errors = {};
errors[2] = "Error message here"
```

* You'll need to change `{data.map((data) => {` to include an index of the current row 
* Then just check if `rowIndex` is defined in the `errors` map. If so, then show an error on that row

You should be able to update the storybook to take in a row number + an error message (or potentially just allow the user to define the object? Not sure how storybook works when you have an input that's an array/an object)

**Comment by gabriellaarcilla** [comment]
we can now specify which rows have errors, and we can display error messages for those rows.

File: FrontEnd/src/components/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                  <tr>
+                    <td>{data.Tag}</td>
+                    <td>{data.Subquadrat}</td>
+                    <td>{data.SpCode}</td>
+                    <td>{data.DBH}</td>
+                    <td>{data.Htmeas}</td>
+                    <td>{data.Codes}</td>
+                    <td>{data.Comments}</td>
+                  </tr>
+                  <tr className="errorMessage">{errorMessage[index]}</tr>
```
Comment (OmarShehata) [inline FrontEnd/src/components/Table.tsx]: If `errorMessage[index]` is not defined, how does it look visually?

Comment (gabriellaarcilla) [reply FrontEnd/src/components/Table.tsx]: Here is what the table looks like without any errors:


Here is what the table looks like with errors on rows 0, 3, and 5

Comment (OmarShehata) [reply FrontEnd/src/components/Table.tsx]: Looks great!

**Comment by OmarShehata** [comment]
This looks good to me - @zmagar @Jaro05 can you review as well and approve if all looks good to you as well?

**Comment by Jaro05** [comment]
This is looking good! Three things, 
1) Is this being hooked up to the actual app in a separate issue? Shouldn't this table appear after a user uploads all of the files? This will probably be a separate issue once we have the upload files button, just want to clarify.
2) It might be a good idea to rename the component to something other than "Table". This app uses a lot of tables so it may get confusing. Maybe ErrrorsTable or something similar.
3) I think we can work on this together even after this PR request once it is hooked up to the app, but maybe adding some CSS including green borders and such to match the style of the app would be good for styling.

But other than that this is great! The use of storybook seems helpful.

**Comment by OmarShehata** [comment]
> Is this being hooked up to the actual app in a separate issue? Shouldn't this table appear after a user uploads all of the files? This will probably be a separate issue once we have the upload files button, just want to clarify.

I think this makes sense to me! Can you please open an issue for this if there isn't one already? 

> It might be a good idea to rename the component to something other than "Table". This app uses a lot of tables so it may get confusing. Maybe ErrrorsTable or something similar.

That's a good idea - maybe "PostUploadTable" since that is when it is displayed. Or actually better "ValidationTable" since it shows the data with any errors, so we can use that not just on post upload, but any time we want to show errors with a specific file.

**Comment by gabriellaarcilla** [comment]
@OmarShehata @Jaro05 I just changed the name Table to ValidationTable in the new commit. Also I created a separate issue to add the ValidationTable to the Validate.tsx page. Please let me know if there is anything else.

**Review by Jaro05 — APPROVED** [review]
Looks good to merge!

**Comment by gabriellaarcilla** [comment]
Yay! Thank you

**Pull request merged**